File: internal/api/dto/tenant.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

type TenantCreateReq struct {
	Name   string `json:"name" validate:"required"`
	Code   string `json:"code" validate:"required"`
	Status int    `json:"status"`
}

type TenantUpdateReq struct {
	Name   *string `json:"name"`
	Status *int    `json:"status"`
}

type TenantResponse struct {
	ID        uuid.UUID `json:"id"`
	Name      string    `json:"name"`
	Code      string    `json:"code"`
	Status    int       `json:"status"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: web/src/api/tenant.ts
import apiClient from './client'

export interface Tenant {
  id: string
  name: string
  code: string
  status: number
  created_at: string
  updated_at: string
}

export interface TenantCreateReq {
  name: string
  code: string
  status?: number
}

export interface TenantUpdateReq {
  name?: string
  status?: number
}

export const tenantApi = {
  list() {
    return apiClient.get<Tenant[]>('/tenants')
  },
  get(id: string) {
    return apiClient.get<Tenant>(`/tenants/${id}`)
  },
  create(data: TenantCreateReq) {
    return apiClient.post<Tenant>('/tenants', data)
  },
  update(id: string, data: TenantUpdateReq) {
    return apiClient.put<Tenant>(`/tenants/${id}`, data)
  },
  delete(id: string) {
    return apiClient.delete(`/tenants/${id}`)
  }
}
